api/c/tests/unit/basic/deco_sdf.cpp

```diff
@@ -0,0 +1,140 @@
+    if (print_molfile)
+    {
+        prepareStructure(full_scaf);
+        std::cout << "full scaffold: " << indigoMolfile( full_scaf ) << std::endl;
+    }
+
+    int decos_it = indigoIterateDecomposedMolecules(deco);
+    while (indigoHasNext(decos_it))
+    {
+        int item = indigoNext(decos_it);
+		std::cout << indigoSmiles( indigoDecomposedMoleculeHighlighted(item) ) << std::endl;
+        int mol = indigoDecomposedMoleculeWithRGroups(item);
+        std::cout << "decomposed molecule: " << indigoCanonicalSmiles(mol) << std::endl;
+        if ( print_molfile )
+	    {
+            prepareStructure( mol );
+            std::cout << "decomposed molecule: " << indigoMolfile(mol) << std::endl;
+		}
+        std::cout << "mapped scaffold: " << indigoCanonicalSmiles(indigoDecomposedMoleculeScaffold(item)) << std::endl;
+        int rgroups = indigoIterateRGroups(mol);
+		while (indigoHasNext(rgroups))
+		{
+            int rg = indigoNext( rgroups);
+            std::cout << "  RGROUP #" << indigoIndex(rg) << std::endl;
+            int frags = indigoIterateRGroupFragments( rg );
+			if (indigoHasNext(frags))
+			{
+                int frag = indigoNext(frags);
+                std::cout << "  fragment #" << indigoIndex(frag) << ":" << indigoCanonicalSmiles(frag) << std::endl;
+            }
+            else
+                std::cout << "NO FRAGMENT" << std::endl;
+		}
+    }
+    sdf_it = indigoIterateSDFile(filename.c_str());
+    while (indigoHasNext(sdf_it))
+    {
+        int item = indigoNext(sdf_it);
+        int sm = indigoSubstructureMatcher( item, nullptr );
+		if (!indigoMatch(sm, full_scaf))
+		{
+            std::cout << "ERROR: full scaffold not found in the input structure " << indigoIndex(item) << std::endl;
+		}
+        int scaffs_it = indigoIterateArray(all_scaffolds);
+        while (indigoHasNext(scaffs_it))
+		{
+            int scaf = indigoNext(scaffs_it);
+            int sm = indigoSubstructureMatcher(item, nullptr);
+            if (!indigoMatch(sm, scaf))
+            {
+                std::cout << "ERROR: scaffold " << indigoIndex( scaf ) << " not found in the input structure " << indigoIndex( item ) << std::endl;
+            }
+		}
+    }
+}
+
+TEST(IndigoDecoSDFTest, deco_sdf_test)
+{
+    qword session = indigoAllocSessionId();
+    indigoSetSessionId(session);
+
+    indigoSetErrorHandler(errorHandling, 0);
+
+    try
+    {
+        testScaffold(dataPath("molecules/basic/thiazolidines.sdf"), "exact 10000", false);
+        testScaffold(dataPath("molecules/basic/thiazolidines.sdf"), "approx", false);
+        testScaffold(dataPath("molecules/basic/thiazolidines.sdf"), "approx 3", false);
+        testScaffold(dataPath("molecules/basic/sugars.sdf"), "exact", true);
+        testScaffold(dataPath("molecules/basic/sugars.sdf"), "approx", false);
+    }
+    catch (Exception& e)
+    {
+        ASSERT_STREQ("", e.message());
+    }
+}
```

Comment:
I don't think we need this test here - C++ unittests should be small and fast. We have this test in integration tests.

Reply:
test removed